src/prototype_creep_resources.js

```diff
@@ -522,20 +522,7 @@ Creep.prototype.getEnergy = function() {
       source = s;
     }
   });
-  if (!source) {
-    if (this.carry.energy > 0) {
-      this.memory.hasEnergy = true; // Stop looking and spend the energy.
-      return false;
-    } else {
-      source = this.pos.findClosestByRange(FIND_SOURCES);
-      this.moveTo(source, {
-        reusePath: 5,
-        ignoreCreeps: true,
-        costCallback: this.room.getAvoids(this.room)
-      });
-      return true;
-    }
-  }
+  source = source || sources[0];
```

Comment:
I can happen that there are no `SOURCES_ACTIVE`.

Maybe we can use in line 515 `FIND_SOURCES`, we select the best one later anyway.